src/chainstate/stacks/index/file.rs

```diff
@@ -392,9 +738,11 @@ impl TrieFile {
         block_id: u32,
         ptr: &TriePtr,
     ) -> Result<TrieHash, Error> {
-        let offset = self.get_trie_offset(db, block_id)?;
-        self.seek(SeekFrom::Start(offset + (ptr.ptr() as u64)))?;
-        let hash_buff = read_hash_bytes(self)?;
+        self.load_trie_blob(db, block_id)?;
```

Comment:
This could easily lead to significant disk thrashing.  In the worst case, you'd be loading a whole trie (10^5 to 10^7 bytes) from disk simply to read a single trie node as part of following backptrs.  Moreover, this evicts other cached trie blobs.

Reply:
This applies to several points in the code where you call `self.load_trie_blob()` as part of reading either a trie node or a trie hash.  For brevity, I won't comment on them all unless you want me to.

Reply:
Hm.. Perhaps maintining a separate index file with all (back)pointers & hashes, which would most likely easily fit in memory as well, could be a potential solution and possibly also offer a performance boost if walking can be done in-memory?

My spontaneous thought anyway.. thoughts?